src/Autofac/Features/Collections/CollectionRegistrationSource.cs

```diff
@@ -160,7 +161,9 @@ public override string ToString()
             var parameter = Expression.Parameter(typeof(int));
             var genericType = typeof(List<>).MakeGenericType(elementType);
             var constructor = genericType.GetMatchingConstructor(new[] { typeof(int) });
-            var newList = Expression.New(constructor, parameter);
+
+            // We know that List<> has the constructor we need.
+            var newList = Expression.New(constructor!, parameter);
```

Comment:
Do we actually know that the `List<T>(int)` constructor is kept once linker trimming is enabled?